lightning/mydump/region.go

```diff
@@ -171,6 +187,48 @@ func MakeTableRegions(
 		dataFileSizes = append(dataFileSizes, float64(dataFileSize))
 	}
 
-	AllocateEngineIDs(filesRegions, dataFileSizes, float64(batchSize), batchImportRatio, float64(tableConcurrency))
+	AllocateEngineIDs(filesRegions, dataFileSizes, float64(cfg.Mydumper.BatchSize), cfg.Mydumper.BatchImportRatio, float64(cfg.App.TableConcurrency))
 	return filesRegions, nil
 }
+
+func splitLargeFile(meta *MDTableMeta, cfg *config.Config, dataFilePath string, dataFileSize int64, divisor int64, prevRowIdxMax int64, ioWorker *worker.Pool) (prevRowIdMax int64, regions []*TableRegion, dataFileSizes []float64, err error) {
+	reader, err := os.Open(dataFilePath)
+	if err != nil {
+		return 0, nil, nil, err
+	}
+	dataFileSizes = make([]float64, 0, dataFileSize/config.MaxRegionSize)
+	parser := NewCSVParser(&cfg.Mydumper.CSV, reader, cfg.Mydumper.ReadBlockSize, ioWorker)
+	startOffset, endOffset := int64(0), config.MaxRegionSize
+	for endOffset < dataFileSize {
+		curRegionCnt := (endOffset - startOffset) / divisor
```

Comment:
```suggestion
		curRowsCount := (endOffset - startOffset) / divisor
```

I've read several times to determine what this variable means 🙃. This counts the maximum possible number of rows between `startOffset` to `endOffset` so that row ID allocations won't overlap. So calling it "region count" is a bit strange.